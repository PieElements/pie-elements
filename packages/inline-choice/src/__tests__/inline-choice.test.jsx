import React from 'react';
import InlineChoice from '../inline-choice';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const props = {
  session: {},
  onChoiceChanged: () => { },
};

const model = {
  choices: [{
    "correct": true,
    "value": "sweden",
    "label": [
      {
        "lang": "en-US",
        "value": "Sweden"
      },
      {
        "lang": "es-ES",
        "value": "Suecia"
      },
      {
        "lang": "zh-CN",
        "value": "瑞典"
      }
    ],
    "feedback": {

    }
  }, {
    "value": "iceland",
    "label": [
      {
        "lang": "en-US",
        "value": "Iceland"
      },
      {
        "lang": "es-ES",
        "value": "Islandia"
      },
      {
        "lang": "zh-CN",
        "value": "冰岛"
      }
    ],
    "feedback": {
      "type": "default",
      "text": [
        {
          "lang": "en-US",
          "value": "Iceland is not an EU member state"
        },
        {
          "lang": "es-ES",
          "value": "Islandia no es un estado miembro de la UE"
        },
        {
          "lang": "zh-CN",
          "value": "冰岛不是欧盟成员国"
        }
      ]
    }
  }],
  classes: {},
  disabled: false
}


describe('InlineChoice Component', () => {
  it('should render component', () => {
    const tree = renderer.create(<InlineChoice
      {...model}
      session={props.session}
      onChoiceChanged={props.onChoiceChanged()} />).toJSON();
    console.log("tree", tree);
    expect(tree).toMatchSnapshot();
  });
});