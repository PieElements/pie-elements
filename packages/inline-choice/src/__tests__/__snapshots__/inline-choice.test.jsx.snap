// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineChoice Component should render component 1`] = `
<div
  className="InlineChoice-container-1"
>
  <div
    className="MuiFormControl-root-4 InlineChoice-formControl-2"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      className="MuiInput-root-13 MuiInput-formControl-14 MuiInput-inkbar-15 MuiInput-underline-19"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiSelect-root-8"
      >
        <div
          aria-haspopup="true"
          aria-owns={null}
          aria-pressed="false"
          className="MuiSelect-select-9 MuiSelect-selectMenu-10 MuiInput-input-22"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={undefined}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        />
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          defaultValue={undefined}
          id="input-choice"
          name="input-choice"
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-28 MuiSelect-icon-12"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
